fdbrpc/FlowTransport.actor.cpp

```diff
@@ -940,6 +972,9 @@ ACTOR static void deliver(TransportData* self,
 			StringRef data = reader.arenaReadAll();
 			ASSERT(data.size() > 8);
 			ArenaObjectReader objReader(reader.arena(), reader.arenaReadAll(), AssumeVersion(reader.protocolVersion()));
+			bool didInsert = objReader.variable<AuthorizedTenants>("AuthorizedTenants", &authorizedTenants);
```

Comment:
If we do this for every packet, it is on performance-relevant path. We could use some numerical IDs instead of strings for identification, also a fixed size array instead of map.

Reply:
Yes, this part needs to be refactored. It's a hack meant to serve as a temporary solution. I should've added a comment though